### PR TITLE
Added AppVeyor validation build configuration

### PR DESCRIPTION
**Pull request by alpaix**
- Ordered .gitattributes settings for text files.
- Normalized the line endings across source files.
- Updated dotnet-test-xunit version to 1.0.0-rc2-173384-19.
- Loosened test build result checks from testing `$?` to simply verifying the `$LASTEXITCODE`.
- Re-wrote GlobalSuppressions.cs files in UTF8.
- Fixed weird dependency of `System.Runtime.4.1.0/net46` by replacing it with framework assembly entry.
- Fixed build warnings.
- Added [Build status](https://ci.appveyor.com/project/NuGetTeam/nuget-client) status to the front repo page.

//cc @drewgil @joelverhagen @emgarten @rrelyea

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
+version: 3.5.0-xprivate-{build}
+pull_requests:
+  do_not_increment_build_number: true
```
Comment (joelverhagen) [inline appveyor.yml:3]: Does this mean each AppVeyor build will have the same build number?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+- git submodule update --init
+build_script:
+- ps: .\build.ps1 -Configuration Release -ReleaseLabel xprivate -BuildNumber $env:APPVEYOR_BUILD_NUMBER -SkipTests -SkipILMerge
+test_script:
+- ps: >-
+    . .\build\common.ps1
+
+    $TestErrors = @()
+
+    Invoke-BuildStep 'Running NuGet.Core tests' {
```
Comment (joelverhagen) [inline appveyor.yml:19]: Perhaps we can add a -SkipBuild option to build.ps1 for this?

File: build/common.ps1 [hunk 3]
```diff
@@ -438,12 +438,14 @@ Function Test-XProject {
...
+                if ($VerbosePreference) {
+                    $opts += '-verbose'
+                }
 
                 Trace-Log "$DotNetExe $opts"
-
                 & $DotNetExe $opts
 
-                if (-not $?) {
+                if ($LASTEXITCODE -ne 0) {
```
Comment (joelverhagen) [inline build/common.ps1:448]: What's the different? I am not familiar with `$?` vs `$LASTEXITCODE`.

File: appveyor.yml (hunk 1 above)
Comment (alpaix) [inline appveyor.yml:3]: this setting is for PR builds only as opposed to on-every-commit dev builds. Copied from other ASP.NET project. Will evaluate it later when actual PR integration is in place.

**Comment by joelverhagen** [comment]
Will this AppVeyor CI be run for all `dev` check-ins? What about PRs? Is this the intended way for community members to get feedback on the quality of their PR?

File: appveyor.yml (hunk 2 above)
Comment (alpaix) [inline appveyor.yml:19]: Prefer custom tailored scripts rather than adding more options. It's hard to keep build.ps1 fitting all possible scenarios.

**Comment by alpaix** [comment]
My plan is to activate "daily" builds and let it run for couple of days. Then work with @rrelyea on setting right permissions for PR builds.

File: appveyor.yml (hunk 2 above)
Comment (joelverhagen) [inline appveyor.yml:19]: I think running tests without a build has general appeal, but your call.

File: build/common.ps1 (hunk 3 above)
Comment (alpaix) [inline build/common.ps1:448]: `$?` is a magic PS variable evaluating outcome of a process run including last exit code, error output, warnings, etc. This introduces a lot of noise on AppVeyor VM. Generally checking last exit code is enough for almost every build tool we use.

**Comment by joelverhagen** [comment]
:shipit:

File: appveyor.yml (hunk 2 above)
Comment (alpaix) [inline appveyor.yml:19]: Actually you gave me good idea. I think having separate test.ps1 script for that will help with that.
Does it make sense?

Comment (joelverhagen) [inline appveyor.yml:19]: Great!

**Pull request merged**